sklearn/decomposition/_nmf.py

```diff
@@ -126,7 +126,7 @@ def _beta_divergence(X, W, H, beta, square_root=False):
         if sp.issparse(X):
             norm_X = np.dot(X.data, X.data)
             norm_WH = trace_dot(np.linalg.multi_dot([W.T, W, H]), H)
-            cross_prod = trace_dot((X * H.T), W)
+            cross_prod = trace_dot((X @ H.T), W)
```

Comment:
Since this PR is actually fixing a problem in the code, we need to document this in changelog entry in `doc/whats_new.rst/v1.4.rst`

The changelog entry should only mention the public classes impacted by this change are `sklearn.decomposition.NMF` and `sklearn.decomposition.MiniBatchNMF` (instead of the private `_beta_divergence` function).